### ISSUE TITLE
May I ask if the model weights uploaded on Hugging Face have been pre trained?

### ISSUE THREAD

**Issue by mop134679852** [360CVGroup/FG-CLIP#6]
Why did I run your demo code and the final **print (probes)** result is (I don't feel the effect is ideal):
**tensor([[0.5269, 0.4731]], device='cuda:0')**
If you have seen it, please help answer it. Thank you.

**Here is my code:**
import torch
from PIL import Image
from transformers import (
    AutoImageProcessor,
    AutoTokenizer,
    AutoModelForCausalLM,
)

model_root = "/data1/fg_clip_large"
image_size=336
model = AutoModelForCausalLM.from_pretrained(model_root,trust_remote_code=True).cuda()

device = model.device

tokenizer = AutoTokenizer.from_pretrained(model_root)
image_processor = AutoImageProcessor.from_pretrained(model_root)

img_root = "/data1/fg_clip_large/cat_dfclor.jpg"
image = Image.open(img_root).convert("RGB")
image = image.resize((image_size,image_size))

image_input = image_processor.preprocess(image, return_tensors='pt')['pixel_values'].to(device)

# NOTE Short captions: max_length=77 && walk_short_pos=True
walk_short_pos = True
captions=["a photo of a cat", "a photo of a dog"]
caption_input = torch.tensor(tokenizer(captions, max_length=77, padding="max_length", truncation=True).input_ids, dtype=torch.long, device=device)

# NOTE Long captions: max_length=248 && walk_short_pos=False
# ......

with torch.no_grad():
  image_feature = model.get_image_features(image_input)
  text_feature = model.get_text_features(caption_input,walk_short_pos=walk_short_pos)
  image_feature = image_feature / image_feature.norm(p=2, dim=-1, keepdim=True)
  text_feature = text_feature / text_feature.norm(p=2, dim=-1, keepdim=True)

logits_per_image = image_feature @ text_feature.T 
probs = logits_per_image.softmax(dim=1) 
print(probs)

**Comment by mop134679852** [comment]
我已经解决了注释掉下面代码就可以了。
image_feature = image_feature / image_feature.norm（p=2， dim=-1， keepdim=True）
text_feature = text_feature / text_feature.norm（p=2， dim=-1， keepdim=True）

**Comment by icech** [comment]
> 我已经解决了注释掉下面代码就可以了。 image_feature = image_feature / image_feature.norm（p=2， dim=-1， keepdim=True） text_feature = text_feature / text_feature.norm（p=2， dim=-1， keepdim=True）

注释完之后，我发现我把猫换成apple，输出等prob是tensor([[0.0178, 0.9822]], device='cuda:0')，离大谱

**Comment by binwang777** [comment]
> 我已经解决了注释掉下面代码就可以了。 image_feature = image_feature / image_feature.norm（p=2， dim=-1， keepdim=True） text_feature = text_feature / text_feature.norm（p=2， dim=-1， keepdim=True）

We checked the test code and found that the temperature coefficient was not multiplied before softmax. We corrected the test code in the readme:

`
with torch.no_grad():
    image_feature = model.get_image_features(image_input)
    text_feature = model.get_text_features(caption_input,walk_short_pos=walk_short_pos)
    image_feature = image_feature / image_feature.norm(p=2, dim=-1, keepdim=True)
    text_feature = text_feature / text_feature.norm(p=2, dim=-1, keepdim=True)

logits_per_image = image_feature @ text_feature.T
logits_per_image = model.logit_scale.exp() * logits_per_image
probs = logits_per_image.softmax(dim=1)
print(probs)
# [[9.9997e-01, 3.3485e-05]]
`
